ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/views/PaywallFooterView.kt

```diff
@@ -85,35 +89,43 @@ open class PaywallFooterView : FrameLayout {
      * Sets the offering id to be used to display the Paywall. If not set, the default one will be used.
      */
     fun setOfferingId(offeringId: String?) {
-        this.offeringId = offeringId
-        invalidate()
+        this.offeringIdState = offeringId
+    }
+
+    /**
+     * Sets the font provider to be used to display the Paywall Footer View.
+     */
+    fun setFontProvider(fontProvider: FontProvider) {
+        this.fontProviderState = fontProvider
     }
 
     private fun init(context: Context, attrs: AttributeSet?) {
         parseAttributes(context, attrs)
         addView(
             ComposeView(context).apply {
                 setContent {
-                    val paywallOptions = PaywallOptions.Builder { dismissHandler?.invoke() }
-                        .setListener(internalListener)
-                        .setFontProvider(fontProvider)
-                        .setOfferingId(offeringId)
-                        .build()
+                    val paywallOptions = remember(fontProviderState, offeringIdState) {
+                        PaywallOptions.Builder { dismissHandler?.invoke() }
+                            .setListener(internalListener)
+                            .setFontProvider(fontProviderState)
+                            .setOfferingId(offeringIdState)
```

Comment:
I may be missing something, but I believe this won't recompute the options on a change on the `offeringIdState`. The inside of the `remember` function is only called once to initialize the value, but it won't listen to state changes AFAIK (Do lmk if I'm wrong!)

Reply:
@tonidero 
so any changes in the state within the key in `remember(key)` should trigger this lambda again with the new PaywallOptions.

Reply:
@Jjastiny From what I understand, the next time it recomposes, it would indeed create a new PaywallOptions instance with the new value due to that `remember(key)`. However, it wouldn't trigger a recomposition by itself since nothing in the compose context is listening to the changes of a `mutableState` (I believe being part of the key of the `remember` does not trigger a recomposition).

Having said that, I haven't tried this code, but I believe the change wouldn't be triggered. Have you been able to confirm the change triggers with this?

Reply:
@tonidero 
yeap ive confirmed that this does indeed cause recomposition; however,the code that was merged to main in ur main branch does not. (It only recomposes once) 😱I think it may cause future bugs, so it would be great if we could start looking at it again!

Reply:
yeah i stand corrected, it looks like it's recomposing in the main branch!